MvvmCross/Platform/Uap/Core/MvxTypedWindowsSetup.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MS-PL license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using MvvmCross.Platform.Uap.Views;
+using MvvmCross.ViewModels;
+using System.Linq;
+using Windows.UI.Xaml.Controls;
+
+namespace MvvmCross.Platform.Uap.Core
+{
+    public class MvxTypedWindowsSetup<TApplication>: MvxWindowsSetup
```

Comment:
I think that having a typed version of the setup could be an idea, but then it should be over all the platforms. Why in particular do you want this? Most people need to have a custom setup anyway. 

For the class, please make the changes to the normal `MvxWindowsSetup` file and add only the generic overrides in there as a new class. Both should have the same name.

Reply:
Ok will change name back - I was having a lot of issues with multi-targetting breaking builds and thought that it was due to it not being able to resolve the generic override :-(